README.md

```diff
@@ -46,19 +46,43 @@ Project documentation is available on the Jetty Eclipse website.
 Building
 ========
 
-To build, use:
+[Apache Maven 3.8.0](https://maven.apache.org/) and [OpenJDK](https://adoptium.net/) requirements:
+
+Branch         | Maven Version | Minimum JDK | Recommended JDK
+---------------|---------------|-------------| ---------------
+`jetty-10.0.x` | Maven 3.8.6+  | OpenJDK 11  | OpenJDK 17 (for optional loom and http/3 support)
+`jetty-11.0.x` | Maven 3.8.6+  | OpenJDK 11  | OpenJDK 17 (for optional loom and http/3 support)
+`jetty-12.0.x` | Maven 3.8.6+  | OpenJDK 17  | OpenJDK 17
```

Comment:
Do we need the recommended JDK column? That might date quickly and I can see us not updating that.

Also, technically we only need to list the mvn and jdk version needed for this branch and this table can(does) live on a webpage that points to the various branches.